OSPSuite Architecture.md

```diff
@@ -0,0 +1,280 @@
+ 
+![Onion Architecture (image from https://jeffreypalermo.com/2008/07/the-onion-architecture-part-2/)](assets/images/onion_architecture_details.png)
+
+
+In this example, the `IConferenceRepository` interface is defined as a DomainServices that can thus be accessed from all presenters of the application. The implementation however resides in Infrastructure.
+This architecture relies HEAVILY on the use of the [Dependency Inversion Principle](http://en.wikipedia.org/wiki/Dependency_inversion_principle) 
+The application core needs implementation of core interfaces, and if those implementing classes reside at the edges of the application, we need some mechanism for injecting that code at runtime so the application can do something useful.
+In OSPSuite, we did not have the need so separate Application Services from Domain Services. We just use services. But the concept remains the same.
+You can refer to the description of the onion architecture from the time of the original design of OSPSuite [part1](https://jeffreypalermo.com/2008/07/the-onion-architecture-part-1/) , [part2](http://jeffreypalermo.com/blog/the-onion-architecture-part-2/) , [part3](http://jeffreypalermo.com/blog/the-onion-architecture-part-3/) and [part4](https://jeffreypalermo.com/2013/08/onion-architecture-part-4-after-four-years/) for more information on this design patterns. Since the adoption of the onion architecture from OSPSuite, it has become ever more popular. You can also refer to the 2022 article about the onion architecture [here](https://medium.com/expedia-group-tech/onion-architecture-deed8a554423).
+
+
+## OSPSuite.Core Project Structure
+
+As discussed above the OSPSuite is mainly divided between three solutions. We will begin with  [OSPSuite Core](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core) that contains all the common functionalities. The OSPSuite.Core solution has been separated in several projects reflecting the different concerns of the application:
+
+![The OSPSuite.Core solution structure](assets/images/ospsuite_core_solution.png)
+
+### OSPSuite.Assets
+
+All strings displayed in the UI (captions, menus, tooltips, errors, warning, messages) and are common to both PK-Sim and MoBi. They are stored in one file call UIConstants. Although not implemented yet, this structure would enable the application to be used in different languages. It is required however to always define new strings in this file, and NEVER in the designer.
+
+### OSPSuite.Assets.Images
+
+In this project we define all the icons and images that are common to both PK-Sim and MoBi.This component has no other dependencies. It provides two static classes, ApplicationIcons and ApplicationImages that can be accessed throughout the presenter or UI layer to display icons and images.
+
+### OSPSuite.Core
```

Comment:
Maybe this section could be bullet list instead of so much unbroken text.

If you implement bullets, then maybe all the other sections should be consistent.

Reply:
@rwmcintosh you mean the individual projects? Or the ApplicationIcons and ApplicationImages?

Reply:
When I look at the document view I see. It looks like there's newlines in the markdown, but they aren't translating to paragraphs. It's a continuous block.

You could introduce bullets, but if so, make everything consistent. Maybe if line breaks are shown it would be better already.


![image](https://github.com/Open-Systems-Pharmacology/developer-docs/assets/261477/4adccf8b-b0ae-44b9-a567-859694100ba4)

Reply:
OK, fixed